lib/src/components/NxTransferList/NxTransferList.tsx

```diff
@@ -38,39 +40,68 @@ export default function NxTransferList<T extends string | number>(props: Props<T
     ...attrs
   } = props;
 
+  if (allowReordering && !Array.isArray(selectedItems)) {
+    throw new TypeError('selectedItems must be an array if allowReordering is true');
+  }
+
+  const selectedItemsArray = Array.from(selectedItems);
+  const selectedItemsSet = selectedItems instanceof Set ? selectedItems as Set<T> : new Set(selectedItemsArray);
+
   const availableItemsCountFormatter = availableItemsCountFormatterProp || defaultAvailableItemsCountFormatter,
       selectedItemsCountFormatter = selectedItemsCountFormatterProp || defaultSelectedItemsCountFormatter;
 
-  const groupedItems = useMemo(() => groupBy(item => selectedItems.has(item.id) ? 'selected' : 'available', allItems),
-          [allItems, selectedItems]),
+  const allItemsIdToItemlookUp = groupBy(item => item.id.toString(), allItems);
+
+  const groupedItems = useMemo(() =>
+        allowReordering
+          ? {
+            available: reject(item => selectedItemsSet.has(item.id), allItems),
+            selected: chain<T, DataItem<T>>(item => allItemsIdToItemlookUp[item.toString()], selectedItemsArray)
```

Comment:
For me it's working without the type annotation

```suggestion
            selected: chain(item => allItemsIdToItemlookUp[item.toString()], selectedItemsArray)
```